src/utils/combo/criteria.ts

```diff
@@ -146,6 +146,35 @@ export const ComboDidKill: Criteria = (combo, _settings, options) => {
   return !options.comboMustKill || combo.didKill;
 };
 
+export const IncludesComboSequence: Criteria = (combo, _, options) => {
+  const sequenceFilter = options.includesComboSequence;
+  const moves = combo.moves.map((move) => move.moveId);
+
+  if (!sequenceFilter || sequenceFilter.length === 0) {
+    return true;
+  }
+  if (moves.length < sequenceFilter.length) {
+    return false;
+  }
+
+  for (let i = 0; i < moves.length - sequenceFilter.length + 1; i++) {
+    if (moves[i] === sequenceFilter[0]) {
+      let match = true;
+      for (let n = 1; n < sequenceFilter.length; n++) {
+        if (moves[i + n] !== sequenceFilter[n]) {
+          match = false;
+          break;
+        }
+      }
+      if (match) {
+        return true;
+      }
```

Comment:
Can you create a generic array comparison function and use that here?
Something along the lines of `isSameArray<T>(arr: T[], other: T[]): boolean`?
I think it will make this function cleaner, and there are likely other places where a generic array comparison function will be useful.

Reply:
The only drawback I can think of with a separate array comparison function is that we would need to use `slice` to create the arrays for comparison which creates a new array each time but I think that shouldn't be too bad and may still be worth it for cleaner code.

Maybe experiement with it and let me know.

Reply:
I can circumvent making a copy if I provide an index in addition to the arrays? Just unsure if that will limit usefulness in other contexts. I'll post a couple snippets and we can discuss the options!

Reply:
I used slice originally and then was worried about performance, but I'll run a benchmark maybe and see if it matters much

Reply:
Okay, so after benchmarking both solutions, it seems that slicing is actually faster than my solution up until about 13k combos calculated or so. You can see the data I collected in the screenshot below.

![image](https://user-images.githubusercontent.com/19626804/206620335-d01f0178-2e54-4d86-81ff-3422b85c81e6.png)

Note that the X axis is a logarithmic scale. I don't think that this will *really* be that much of a difference practically speaking, and the cleanliness gained is worth it. What do you think of this solution?

```ts
export const IncludesComboSequence: Criteria = (combo, _, options) => {
  const sequenceFilter = options.includesComboSequence;
  const moves = combo.moves.map((move) => move.moveId);

  if (!sequenceFilter || sequenceFilter.length === 0) {
    return true;
  }
  if (moves.length < sequenceFilter.length) {
    return false;
  }

  for (let i = 0; i < moves.length - sequenceFilter.length + 1; i++) {
    // I feel like this conditional is a bit lengthy but I want it to short circuit
    if (moves[i] === sequenceFilter[0] && IsEquivalentArray(moves.slice(i, i + sequenceFilter.length), sequenceFilter)) {
      return true;
    }
  }

  return false;
};
```
Where `IsEquivalentArray<T>(a: T[], b: T[]): boolean` is defined in `src/utils/helpers.ts` (???) as
```ts
export const IsEquivalentArray = <T>(arrA: T[], arrB: T[]): boolean => {
  if (arrA.length !== arrB.length) return false;
  for (let i = 0; i < arrA.length; i++) {
    if (arrA[i] !== arrB[i]) {
      return false;
    }
  }
  return true;
};
```

Also I used "Is**Equivalent**Array" just to not confuse with checking for the arrays to *reference* the same actual array.

Reply:
This looks fine! I'm happy with it :)
Would prefer the declaration to be `export function isEquivalentArray<T>(arrA: T[], arrB: T[]): boolean {}` though but yeah it's looking really good. :)

Reply:
I could do function instead of arrow if you'd like, but all of the other functions in helpers.ts, criteria.ts, and filter.ts use arrow functions and I was matching those for consistency. Would you still rather I use `function`?

Reply:
ohhhh really? I haven't looked at the file in a long time xD in that case maybe keep the consistency but just change the function name to be starting with a lower case.

Reply:
Fixed the casing